src/linear_algebra/vandermonde.lean

```diff
@@ -133,4 +134,63 @@ lemma det_vandermonde_ne_zero_iff [is_domain R] {n : ℕ} {v : fin n → R} :
   det (vandermonde v) ≠ 0 ↔ function.injective v :=
 by simpa only [det_vandermonde_eq_zero_iff, ne.def, not_exists, not_and, not_not]
 
+theorem vandermonde_invertibility' {R : Type*} [comm_ring R]
+[is_domain R] {n : ℕ} (v : fin n ↪ R) {f : fin n → R}
+(h₂ : ∀ j, ∑ i : fin n, (v j ^ (i : ℕ)) * f i = 0) : f = 0
+:= eq_zero_of_mul_vec_eq_zero (det_vandermonde_ne_zero_iff.mpr v.inj') (funext h₂)
+
+theorem vandermonde_invertibility {R : Type*} [comm_ring R]
+[is_domain R] {n : ℕ} (v : fin n ↪ R) {f : fin n → R}
```

Comment:
Why is `v` explicit but `f` implicit? I feel like due to `h₂`, it should either be both implicit or both explicit.

Reply:
It turns out that when you're using this in practice, often the f is clear from context but it needs help finding the v. I'd be happy to make both explicit if you think that's more to style.

Reply:
It's not a big deal, just a thing to keep in mind.